src/erlfmt_cli.erl

```diff
@@ -84,38 +94,97 @@ do_unprotected(Opts, Name) ->
     end.
 
 format_file(FileName, Config) ->
-    #config{pragma = Pragma, width = Width, verbose = Verbose, out = Out} = Config,
+    #config{pragma = Pragma, width = Width, verbose = Verbose, check = Check, out = Out} = Config,
     case Verbose of
         true -> io:format(standard_error, "Formatting ~s\n", [FileName]);
         false -> ok
     end,
     Options = [{pragma, Pragma}] ++ [{width, Width} || Width =/= undefined],
-    case erlfmt:format_file(FileName, Out, Options) of
-        {ok, Warnings} ->
+    Result =
+        case {Check, FileName} of
+            {true, stdin} ->
+                check_stdin(Options);
+            _ ->
+                erlfmt:format_file(FileName, Options)
+        end,
+    case {Verbose, Result} of
+        {true, {skip, _}} ->
+            io:format(standard_error, "Skipping ~s because of missing @format pragma\n", [FileName]);
+        _ ->
+            ok
+    end,
+    case Result of
+        {ok, FormattedText, Warnings} ->
             [print_error_info(Warning) || Warning <- Warnings],
-            ok;
-        {check_failed, _, _, Warnings} ->
+            case Check of
+                true -> ok;
```

Comment:
Do we actually perform the check now? I can't see when it is done

Reply:
We performed check on stdin and not on files.
I fixed this now
And reran the test plan properly this time.
Good catch, thank you